### ISSUE TITLE
`Enhancement` Add support for climate preset modes

### ISSUE THREAD

**Issue by lordcoxis** [Blackymas/NSPanel_HA_Blueprint#2394]
### Enhancement Summary

Add support for climate preset modes

### Detailed Description

As of v4.3.12 climate preset modes are not supported, only HVAC modes. There are some climate devices (and integrations like [Versatile Thermostat](https://github.com/jmcollin78/versatile_thermostat)) that use preset modes and so it would be nice to support them.

**Comment by edwardtfn** [comment]
This is something I wanna work on soon as there are many similar requests:
- #1186 
- #1284 
- #1702 

This will require support to a "select" page:
- #1443 
- #1780

**Comment by drpixel972** [comment]
Hello,

@edwardtfn : A select button may or may not be needed.
I think that the same layout as the hvac mode (icon representing the preset) may be more user friendly.
But I don't know if it is easier to code with the select, or the icons ^^.

@lordcoxis : I'm in the same situation (vtherm and nspanel). What I may do, is an automation that select the right preset from the selected temperature.

Best Regards,

Christophe

**Comment by edwardtfn** [comment]
You're right, we probably could solve this with icons...
Are those icons for preset mode standardized?

**Comment by lordcoxis** [comment]
I agree with using the same layout as HVAC modes (icons), that's what I use in regular HA dashboards anyway. In Versatile Thermostat we have the following presets:

Activity: mdi:motion-sensor
Eco: mdi:leaf
Comfort: mdi:sofa
Boost: mdi:rocket-launch
Manual: mdi:knob
Frost: mdi:snowflake

Ideally one could select which ones to show (I only use activity, eco, comfort and boost for example), but that's for another day I guess.

**Comment by drpixel972** [comment]
I don't think that the icons are standardized, as the presets themselves are not (even though most of them are very common).
Two possibles situations : 
- Focus on the most common one, and specify the icons which are common in the lovelace cards.
- More universal, create a dictionnary with the preset and the associated icon (the blueprint then, should help the user to create the dictionnary)

**Comment by lordcoxis** [comment]
@edwardtfn any progress on this? :)

**Comment by lordcoxis** [comment]
Winter is coming so I'm respectfully bumping this ;)

**Comment by edwardtfn** [comment]
Not stale.
I still wanna look at this.
